.github/workflows/build.yml

```
@@ -0,0 +1,109 @@
+name: On Pull Request
+
+on:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    env:
+      php-version: 8.4
+      extensions: grpc, protobuf
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Commitlint and Other Shared Build Steps
+        uses: momentohq/standards-and-practices/github-actions/shared-build@gh-actions-v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set cache key
+        id: cache-key
+        run: echo "key=php-${{ env.php-version }}" >> $GITHUB_OUTPUT
+
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
+        with:
+          php-version: ${{ env.php-version }}
+          extensions: ${{ env.extensions }}
+          key: ${{ steps.cache-key.outputs.key }}
+
+      - name: Cache extensions
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.extcache.outputs.dir }}
+          key: ${{ steps.extcache.outputs.key }}
+          restore-keys: ${{ steps.extcache.outputs.key }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ env.php-version }}
+          extensions: ${{ env.extensions }}
+          tools: composer
+
+      - name: Install dependencies
+        run: composer install
+
+      - name: Check CS (Code Style)
+        run: composer lint
+
+  test:
+    runs-on: ubuntu-latest
+    services:
+      redis:
+        image: redis:latest
+        ports:
+          - 6379:6379
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set cache key
+        id: cache-key
+        run: echo "key=php-8.4-backend" >> $GITHUB_OUTPUT
+
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
+        with:
+          php-version: 8.4
+          extensions: grpc, protobuf
+          key: ${{ steps.cache-key.outputs.key }}
+
+      - name: Cache extensions
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.extcache.outputs.dir }}
+          key: ${{ steps.extcache.outputs.key }}
+          restore-keys: ${{ steps.extcache.outputs.key }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.4
+          extensions: grpc, protobuf
+          tools: composer
+
+      - name: Install dependencies
+        run: composer install
+
+      - name: Run tests with Momento backend
+        env:
+          MOMENTO_API_KEY: ${{ secrets.ALPHA_TEST_AUTH_TOKEN }}
+          TEST_REDIS: false
+        run: |
+          echo "Running tests with Momento backend"
+          composer test
+
+      - name: Run tests with Redis backend
+        env:
+          TEST_REDIS: true
+        run: |
+          echo "Running tests with Redis backend"
+          echo "TEST_REDIS=true" >> $GITHUB_ENV
+          composer test
```

```suggestion
          composer test
```

The echo to `$GITHUB_ENV` isn't necessary since you set it in the test env.